### ISSUE TITLE
[webkit-debug-adapter] Path must be a string. Received undefined

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode-chrome-debug#117]
_From @hansrwindhoff on March 8, 2016 0:15_

Debugging with chrome debugger extension fails with:
[webkit-debug-adapter] Path must be a string. Received undefined

this is the launch.json
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Launch index.html",
			"type": "chrome",
			"request": "launch",
			"file": "${workspaceRoot}/index.html"
		},
		{
			"name": "Launch chrome with localhost:3000, you must start serve seperately",
			"type": "chrome",
			"request": "launch",
			"url": "http://localhost:3000",
			"sourceMaps": true,
			"webRoot": ".",
            "file": ""
		},
		{
			"name": "Attach",
			"type": "chrome",
			"request": "attach",
			"port": 9222
		}
	]
}

This used to work just fine until the 0.10.10 update.

_Copied from original issue: Microsoft/vscode#3792_

**Comment by weinand** [comment]
_From @anwalkers on March 8, 2016 1:59_

+1 Same here...

![image](https://cloud.githubusercontent.com/assets/3085122/13589623/0fa8b2fe-e48e-11e5-9b1d-766b04d5abf3.png)

![image](https://cloud.githubusercontent.com/assets/3085122/13589637/3c18c2de-e48e-11e5-8eb0-b6754882f981.png)

`{
    "version": "0.1.0",
    // "debugServer": "4712",
    "configurations": [
        {
            "name": "test chrome",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:8080/index.html",
            "sourceMaps": true,
            "diagnosticLogging": true,
            "webRoot": "wwwroot"
        },
        {
            "name": "launch for file",
            "type": "chrome",
            "request": "launch",
            "file": "wwwroot/index.html",
            "sourceMaps": true,
            "diagnosticLogging": true,
            "webRoot": "wwwroot/out/client with space"
        },
        {
            "name": "attach to chrome",
            "type": "chrome",
            "port": 9222,
            "request": "attach",
            "sourceMaps": true,
            "diagnosticLogging": true,
            "webRoot": "./wwwroot"
        }
    ]
}
`

Stack from developer tools:

`messageService.ts:147 [webkit-debug-adapter] Path must be a string. Received undefined: Error: [webkit-debug-adapter] Path must be a string. Received undefined
    at Object.E [as create] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:27:15793)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:98:24744
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:17245)
    at Object.n.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20573)
    at n.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21894)
    at n.undefined.t.Class.define._chainedError (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21253)
    at n (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:19081)
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:17245)
    at Object.n.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20573)
    at n.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21894)e.doShow @ messageService.ts:147e.show @ messageService.ts:131(anonymous function) @ compositePart.ts:294e._emitToSpecificTypeListeners @ eventEmitter.ts:241e._emitEvents @ eventEmitter.ts:267e.emit @ eventEmitter.ts:285(anonymous function) @ actions.ts:288p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012n.undefined.t.Class.define.then @ winjs.base.raw.js:1694n.undefined.O.enter @ winjs.base.raw.js:1023n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._setCompleteValue @ winjs.base.raw.js:1327p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._chainedError @ winjs.base.raw.js:1306n @ winjs.base.raw.js:1012n.undefined.t.Class.define.then @ winjs.base.raw.js:1694n.undefined.O.enter @ winjs.base.raw.js:1023n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._setCompleteValue @ winjs.base.raw.js:1327p @ winjs.base.raw.js:1470n.undefined.F.enter @ winjs.base.raw.js:1176n.undefined.t.Class.define._run @ winjs.base.raw.js:1343n.undefined.t.Class.define._error @ winjs.base.raw.js:1316(anonymous function) @ v8Protocol.ts:79t.dispatch @ v8Protocol.ts:147t.handleData @ v8Protocol.ts:114(anonymous function) @ v8Protocol.ts:69emitOne @ events.js:77emit @ events.js:169readableAddChunk @ _stream_readable.js:146Readable.push @ _stream_readable.js:110onread @ net.js:523`

**Comment by weinand** [comment]
_From @MichaelPetrinolis on March 8, 2016 7:0_

I uninstalled extension and reinstalled, same thing. Version 0.10.10 seems to break something

**Comment by mikkelkaas** [comment]
+1
Same bug here - makes it impossible to debug my project
`{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Attach to chrome",
            "type": "chrome",
            "request": "attach",
            "port": 9222,
            "webRoot": "./app"
        }
    ]
}
`
![snipimage](https://cloud.githubusercontent.com/assets/57084/13598631/3148267e-e51f-11e5-96e0-9ea3883bfdcf.JPG)

**Comment by weinand** [comment]
@mikkelkaas as a workaround you might want to try to set the attribute `outDir` to the path where the generated JavaScript lives e.g. `${workspaceRoot}/out`.

**Comment by marko-baric-travcorp** [comment]
I am having the issue after I updated VS Code this morning. 
Basically, I can only start debugger when "webRoot" value starts with "/" (e.g. /app/dis). or it's an absolute path. VS Code version is 0.10.10.

Maybe related to this, "sourceMaps" is not working at all - I cannot make VS Code to stop on the breakpoint in the original source file, only on minifed js file.

**Comment by jcramer** [comment]
+1, same here

**Comment by mikkelkaas** [comment]
I managed to get it working by replacing `{ "version": "0.2.0", "configurations": [ { "name": "Attach to chrome", "type": "chrome", "request": "attach", "port": 9222, "webRoot": "./app" } ] }`
with 
`{ "version": "0.2.0", "configurations": [ { "name": "Attach to chrome", "type": "chrome", "request": "attach", "port": 9222, "webRoot": "${workspaceRoot}/app" } ] }`

**Comment by MichaelPetrinolis** [comment]
I can also confirm the workaround, 
prefixing value in webRoot with ${workspaceRoot}

**Comment by MartinMa** [comment]
Yes, it's a breaking change in 0.10.10 (February 2016). Only absolute paths seem to be allowed in `launch.json` from now on. See the changelog here: https://code.visualstudio.com/Updates#_debugging

I made a pull request to update the defaults and parameter info: https://github.com/Microsoft/vscode-chrome-debug/pull/118

**Comment by joshcothran** [comment]
Figured this out last night as well after much heartburn. @MartinMa @MichaelPetrinolis @weinand thanks for jumping on this.

**TLDR;
Prefix the webRoot attribute with `${workspaceRoot}/`, e.g.:
- 'webRoot: ${workspaceRoot}' (for ".")
- 'webRoot: ${workspaceRoot}/subdir' (for a subdirectory)
**

Frome the [0.10.10 Release Notes: Debugging](https://code.visualstudio.com/Updates#_debugging):
> ### VS Code no longer 'fixes' Relative Paths in Launch Configurations
> In the January milestone, we deprecated the use of relative paths in launch configurations (but still continued to convert relative in absolute paths). With this milestone, we've dropped this automatic fixing in favor of a more transparent strategy: VS Code no longer modifies launch configuration paths when sending them to the debug adapter. This is now the same treatment of paths as for task configurations. ... **Just prefixing the relative path with a `${workspaceRoot}/` should fix the problem.**

**Comment by auchenberg** [comment]
Thanks for the investigation @joshcothran , @MartinMa, @MichaelPetrinolis and @weinand 

PR has been merged in, and the README is now updated with the new defaults.

**Comment by oksurf** [comment]
Used to all work fine without the path..,  but now, adding the path like this:
{
	"name": "Launch Chrome",
	"type": "chrome",
	"request": "launch",
        "file": "${workspaceRoot}/index.html",
...}

I get:
XMLHttpRequest cannot load file:///C:/Users/.../Angular2/quickstart/app/main.js. Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https, chrome-extension-resource.
undefined

**Comment by joshcothran** [comment]
@oksurf This sounds like a different (but possibly related) error; you might consider opening a separate issue and referencing this one. It likely has to do with your web server configuration. Have a look at #121 [here](https://github.com/Microsoft/vscode-chrome-debug/issues/121#issuecomment-195031465); that error also came up for @maxhor as he was troubleshooting. He also pointed out a StackOverflow post - [AngularJS Error: Cross-origin requests are only supported for protocol schemes...](http://stackoverflow.com/questions/27742070/angularjs-error-cross-origin-requests-are-only-supported-for-protocol-schemes) - on this error more generally.
